Verify Translation in Development

imajit opened the pull request:
Fixes #9686 
To review of FTOs easily, now the contributor can quickly change the required files and run it on Gitpod. As some of the FTOs related to translation need UI checks, in PR contributors can add the Gitpod UI screenshots.
![ezgif com-gif-maker](https://user-images.githubusercontent.com/38528640/124307955-2c48c800-db86-11eb-843c-5a69f11fc890.gif)
I read about this method to show the button in **development mode only**, not sure if this is the best practice, also some numbers are attached to the link whenever the button is pressed, it doesn't affect the result though not sure how it is getting added each time.

noi5e commented:
@imajit Nice work!

imajit commented:
The tests commit fails , is this because the code the tests are checking is not there in the code-base yet ?

jywarren commented on app/controllers/home_controller.rb:
```diff
@@ -47,6 +47,16 @@ def dashboard
     end
   end
 
+  def switch_on_translation
```
maybe we should name this
```suggestion
  def switch_on_translation_helper
```

Is this only for development mode? If so, couldn't we use an existing route somehow -- https://github.com/publiclab/plots2/blob/main/test/functional/user_tags_controller_test.rb#L8-L15 maybe? To add the tag? though i guess it's 2 steps.

jywarren commented:
> The tests commit fails , is this because the code the tests are checking is not there in the code-base yet ?

If the code is not there yet, then yes! But you can then add the code and the test would then begin to pass. That would be "Test Driven Development" - a nice pattern!

jywarren commented on config/routes.rb:
```diff
@@ -378,6 +378,8 @@
   post 'comment/react/delete/:id' => 'comment#react_delete'
   post 'comment/react/update/:id' => 'comment#react_update'
 
+  get 'switch_on_translation' => 'home#switch_on_translation'
```
Yes, if possible if we could use existing routes, we'd be able to skip some of the tests because we wouldn't be introducing a whole new route, you know? That would keep codebase complexity down a bit, which is really helpful in the long run on a big project like this.

imajit commented:
![ezgif com-gif-maker(1)](https://user-images.githubusercontent.com/38528640/125864128-9f9a8cb4-9938-4613-b4f9-17ef2c98caa3.gif)
I initially kept this button in the welcome message I could not see it anymore so I just placed it in the top part. I added an extra parameter to the existing route and changed the controller so now no new routes or controllers are needed. Thanks @jywarren for the suggestion it was a bit more complex but yes it keeps the codebase simple now  :smile: . I'm not sure why codecov is failing though for codeclimate I guess I'll add a commit later.
@noi5e @jywarren should I move the button to some other place ?

jywarren commented:
This is looking great! As to another place, it's only in development mode, but maybe it'd confuse people? What if we put it in the navbar menu? And don't worry about CodeCov, i'm working on that in https://github.com/publiclab/plots2/pull/9905

Thanks!

imajit commented:
The top Nav-bar right ? the one with the search bar

Tlazypanda commented:
Ohh @imajit can't say for sure but I think yes @jywarren meant the top nav bar :sweat_smile: looping him here in case I missed it

imajit commented:
This is how it looks in top-nav 
![ezgif com-gif-maker(2)](https://user-images.githubusercontent.com/38528640/126032897-5c9367d2-3152-4d49-b1da-fc8a7db23128.gif)

jywarren commented:
Would it be possible to put it in the same dropdown as under the profile
picture (the small grey circle to the right side)? Thank you!

On Sat, Jul 17, 2021 at 5:43 AM Ajit Mujumdar ***@***.***>
wrote:

> This is how it looks in top-nav
> [image: ezgif com-gif-maker(2)]
> <https://user-images.githubusercontent.com/38528640/126032897-5c9367d2-3152-4d49-b1da-fc8a7db23128.gif>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/9878#issuecomment-881868892>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAF6JZRVCBPZNMRQUUC77DTYFGFXANCNFSM47XCTCRQ>
> .
>

imajit commented:
![ezgif com-gif-maker(3)](https://user-images.githubusercontent.com/38528640/126059675-e6b51a04-32a2-4415-8d2f-b1aa640af50c.gif)

noi5e commented:
Looks good to me! @jywarren, the `Switch on Translation` button is now in the nav, so I believe this one is merge-ready

jywarren commented:
Excellent, thank you so much!!!
